pkg/vm/memory/memory_test.go

```diff
@@ -276,3 +277,78 @@ func TestMemorySegmentsLoadDataTwoElements(t *testing.T) {
 		t.Errorf("Inserted value and original value are not the same")
 	}
 }
+
+func TestValidateExistingMemoryForRangeCheckWithinBounds(t *testing.T) {
+	ratio := uint32(8)
+	builtin := builtinrunner.NewRangeCheckBuiltinRunner(&ratio, 8, true)
+	segments := memory.NewMemorySegmentManager()
+	builtin.InitializeSegments(&segments)
+	builtin.AddValidationRule(&segments.Memory)
+
+	for i := 0; i < 3; i++ {
+		segments.AddSegment()
+	}
+	addr := memory.NewRelocatable(0, 0)
+	val := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromUint64(45))
+	segments.Memory.Insert(addr, val)
+	segments.Memory.ValidateAddress(addr)
+	if !segments.Memory.ValidatedAdresses.Contains(addr) {
+		t.Errorf("Memory failed validating addresses within bounds")
+	}
+
+}
+
+// func TestValidateExistingMemoryForRangeCheckOutsideBounds(t *testing.T) {
+// 	ratio := uint32(8)
+// 	builtin := builtinrunner.NewRangeCheckBuiltinRunner(&ratio, 8, true)
+// 	segments := memory.NewMemorySegmentManager()
+// 	builtin.InitializeSegments(&segments)
+// 	builtin.AddValidationRule(&segments.Memory)
+
+// 	for i := 0; i < 3; i++ {
+// 		segments.AddSegment()
+// 	}
+// 	addr := memory.NewRelocatable(1, 0)
+// 	val := memory.NewMaybeRelocatableFelt(lambdaworks.FeltFromDecString("5"))
+// 	segments.Memory.Insert(addr, val)
+// 	err := segments.Memory.ValidateAddress(addr)
+// 	if err.Error() != "RangeCheckNumOutOfBounds" {
+// 		t.Error("Should fail with RangeCheckNumOutOfBounds")
+// 	}
+// }
```

Comment:
Does this test work? Is it necessary? If not, we can remove this one

Reply:
Fixed test